### DISCUSSION TITLE
Best practice for organizing & grouping huge amount of SSH profiles?

### DISCUSSION THREAD

**Discussion by MichaelSuen-thePointer** [microsoft/terminal#16167] [Q&A]
Hi all. As a server admin, I need to manage multiple server clusters, each cluster consists of multiple machines, from 4 to 500+ machines. In total I have about 8000 ssh profiles to store & access in Terminal.

I can enumerate each machine's ip, name and its belonging cluster. I already have a config generator program to generate those profiles into Terminal's settings.json.

My question is:

I need those ssh profile to be grouped by its cluster names for quicker access.
I need to do fuzzy search on cluster/machine names.
I don't need to open lot's of ssh sessions at the same time.

 Which feature should I use to achieve that?

**Comment by zadjii-msft** [comment]
Maybe something like [New tab dropdown](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/appearance#new-tab-dropdown) customization? You may still need to roll the customization manually (or with a script), but at least they'll all be grouped together in the UI?

To get something like:
![image](https://github.com/microsoft/terminal/assets/18356694/f6d2e762-0bdf-4314-b52a-252fc69a4f43)

I have something like: 

```json
"newTabMenu":
[
    { "type": "remainingProfiles" },
    {
        "allowEmpty": false,
        "entries":
        [
            {
                "commandline": null,
                "name": null,
                "source": "Windows.Terminal.VisualStudio",
                "type": "matchProfiles"
            },
            {
                "type": "profile", "profile": "{da29b8a9-e007-5b35-a0d3-d7a869ae486e}",
            }
        ],
        "icon": null,
        "inline": "never",
        "name": "VS",
        "type": "folder"
    },
]
```

**Reply by MichaelSuen-thePointer** [reply]
Viable solution! The only problem is the tab entries (VS, WSLs, Canonical, etc. which in my case will be server cluster names) cannot be searched (maybe in ctrl+shift+p)
